packages/sources/tradermade-test/src/endpoint/forex-router.ts

```diff
@@ -0,0 +1,21 @@
+import { RoutingTransport } from '@chainlink/external-adapter-framework/transports/meta'
+import { PriceEndpoint } from '@chainlink/external-adapter-framework/adapter'
+import { BatchEndpointTypes } from '../price-utils'
+import { httpTransport, inputParameters } from './forex'
+import { wsTransport } from './forex-ws'
+
+const transports = {
+  ws: wsTransport,
+  rest: httpTransport,
+}
+
+export const routingTransport = new RoutingTransport<BatchEndpointTypes>(transports, {
+  defaultTransport: 'rest',
+})
+
+export const endpoint = new PriceEndpoint<BatchEndpointTypes>({
+  name: 'forex',
+  aliases: ['batch'],
+  transport: routingTransport,
+  inputParameters: inputParameters,
+})
```

Comment:
Shoot sorry I keep realizing things when I'm doing my "final" pass. Since the bootstrap validation is removed, I think we need to add a new one to make sure `ws` requests are rejected if `WS_API_KEY` isn't set. Since the configuration is at the request level now, we can use a `customInputValidation` for this endpoint instead. I think `ncfx` should have an example on its forex endpoint